zerver/models/groups.py

```diff
@@ -37,7 +37,7 @@ class UserGroup(models.Model):  # type: ignore[django-manager-missing] # django-
     description = models.TextField(default="")
     is_system_group = models.BooleanField(default=False)
 
-    can_mention_group = models.ForeignKey("self", on_delete=models.RESTRICT)
+    can_mention_groups = models.ManyToManyField("self", symmetrical=False)
```

Comment:
I have got a couple of questions here -
- Do we need a backwards relation here to get which group uses this groups as `can_mention_groups` and do we want to name it ourselves?
- I am not sure whether we should using `through` table defined by us here because we would need to define a separate table for each setting. There are a couple of places where we could use the through table directly for efficient queries and for that we can use `UserGroup.can_mention_groups.through` including the import-export code.

Reply:
I think there will be situations where we want to query what groups a given group has permissions on; I think a backwards relation can be defined later though as it's not a real thing in the database, just a Django helper.

The big question in my mind in reading this is whether we should instead be making a single `ManyToManyField` table containing a pair of group IDs and a `setting_name` field that equals `can_mention_group`... such that we would not need to do future database format migrations at all in order to add new settings of this form -- just start storing rows of with a new `setting_name` in that single permissions table.

I'm not sure, but I think with the right indexes, that model would be just as fast as what this PR proposes, and might be significantly less annoying to adjust over time? I guess a downside would be that it'd be harder to use `prefetch_related` to fetch only a single mapped group when fetching an object. Probably worth discussing in chat.zulip.org?

Reply:
I actually tried this before but we cannot have multiple `ManyToManyField` using the same throuh table and django gives an error `The model has two identical many-to-many relations through the intermediate model 'zerver.UserGroupPermissionSetting'`.

Reply:
Yeah, so I think we'd just have to do our own `UserGroupPermissions` table without using the `ManyToManyField` infrastructure. I'm not sure to what extent we're really using `ManyToManyField` for much: I guess the big thing would be that it'd be harder to write code to prefetch the permissions rows for a specific permission. Let's discuss in #backend.

Reply:
Yes, we can do that. Would start a discussion.